### DISCUSSION TITLE
Retry Wamp connection when it fails

### DISCUSSION THREAD

**Discussion by berTrindade** [crossbario/autobahn-js#577] [Q&A]
Hi @oberstet, how are you sir ?

I've been checking out the `autobahn-js` repo and seeking a way to retry the Wamp connection in case it fails. But after doing that and reading Crossbar's doc, I didn't see any part where that could be achieved.  

I want to be able to change both `retry_count` and `will_retry` values. But not sure if that'll help.

Would you know if there's a way to do that ? 

Best regards,

**Comment by oberstet** [comment]
please have a look at "automatic reconnection": https://github.com/crossbario/autobahn-js/blob/master/doc/reference.md#connection-options

**Comment by berTrindade** [comment]
@oberstet, I've already have the -1 set to `max_retry`, so AFAIK suppose to be unlimited the number of retries. 

The error I'm facing is the one from the image below:

`WAMP-CRA signature is invalid` and `wamp.error.not_authorized`. I tracked it down and it's happening on this statement: `autobahn.auth_cra.sign` on `onchallenge`.

<img width="780" alt="image" src="https://github.com/crossbario/autobahn-js/assets/13338262/a9b80573-0f64-4ce5-90fd-5e81e32b2c56">

**Comment by oberstet** [comment]
Well, it seems to indicate an authentication problem - which seems unrelated to auto-reconnect. AutobahnJS will fully reauthenticate for every reconnect. A connection is "reconnected" by creating a fresh connection .. however, there is no language to differentiate between "reconnecting == automatically establish a new connection" and "reconnecting = recover a closed connection by reopening that connection" - the latter doesn't exist, the former is what is meant with "reconnecting"

**Comment by berTrindade** [comment]
@oberstet But what would you suggest for a case where I need to recover from a closed connection by creating a new connection ?

**Comment by oberstet** [comment]
no, actually, what you can see from the log is: the client indeed reconnects, but the authentication fails. if the first connect succeeds, but follow ups fail, it could be a client side (your code) bug ... are you recomputing a new signature for the follow up connections (WAMP-CRA has replay protection)

**Comment by oberstet** [comment]
you can also try to use WAMP-Anon -- run without WAMP-CRA authentication and assure yourself that in this case the automatic reconnect works (eg when you reboot your router). bisect the problem.

**Comment by berTrindade** [comment]
@oberstet Thank you. We'll handle it with client retry.
